src/NuGet.Core/NuGet.Packaging/Rules/NoRefOrLibFolderInPackageRule.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Rules
+{
+    class NoRefOrLibFolderInPackageRule : IPackageRule
+    {
+        public string MessageFormat { get; }
+
+        public NoRefOrLibFolderInPackageRule(string messageFormat)
+        {
+            MessageFormat = messageFormat;
+        }
+
+        public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader package)
+        {
+            var files = package.GetFiles();
+            if (!files
+                    .Select(t => PathUtility.GetPathWithDirectorySeparator(t))
+                    .Any(t => t.StartsWith(PackagingConstants.Folders.Lib + Path.DirectorySeparatorChar, StringComparison.OrdinalIgnoreCase)
+                    || t.StartsWith(PackagingConstants.Folders.Ref + Path.DirectorySeparatorChar, StringComparison.OrdinalIgnoreCase)))
+            {
+                //if you can't find the ref and lib folder, then find the build folder
+                if (files
+                    .Select(t => PathUtility.GetPathWithDirectorySeparator(t))
+                    .Any(t => t.StartsWith(PackagingConstants.Folders.Build + Path.DirectorySeparatorChar, StringComparison.OrdinalIgnoreCase)))
+                {
+                    //if you can't find any folders other than native or any, raise an NU5127
```

Comment:
shouldn't it be the other way around?

If the package contains only build/native files, we shouldn't warn because native packages don't work the same way as .NET packages. If the package contains only build/any files, we shouldn't warn because the package is intentionally matching all TFMs. Everything else should warn because the logic in `CompatibilityChecker.cs` isn't going to match packages as the package author probably expects.

Reply:
So the logic is that, the comment hadn't been updated to reflect that